### PR TITLE
Honor envs properly for access and secret key.

### LINKED ISSUE #3696: admin: mc admin service creds changes password for play
```
mc admin service creds play minio minio123
mc admin service creds --debug play minio minio123

Host: play.minio.io:9000
User-Agent: Minio (linux; amd64) madmin-go/0.0.1 mc/2017-02-07T01:29:50Z
Content-Length: 82
Authorization: AWS4-HMAC-SHA256 Credential=**REDACTED**/20170207/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date;x-minio-operation, Signature=**REDACTED**
Content-Md5: lm/6ed9K2gGyAsshRTjBvw==
X-Amz-Content-Sha256: UNSIGNED-PAYLOAD
X-Amz-Date: 20170207T013439Z
X-Minio-Operation: set-credentials
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 403 Forbidden
Transfer-Encoding: chunked
Accept-Ranges: bytes
Content-Type: application/xml
Date: Tue, 07 Feb 2017 01:34:40 GMT
Server: Minio/DEVELOPMENT.2017-02-02T20-07-53Z (linux; amd64)
Vary: Origin
X-Amz-Request-Id: 14A0DD5FB03756D8

111
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>InvalidAccessKeyID</Code><Message>The access key ID you provided does not exist in our records.</Message><Key></Key><BucketName></BucketName><Resource>/</Resource><RequestId>3L137</RequestId><HostId>3L137</HostId></Error>
``

## Expected Behavior
We shouldn't allow change of password for server if it has been started with custom ENVs for ACCESS and SECRET key.
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->

## Current Behavior
Changes the password without validation.
<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->

## Possible Solution
Disallow change in password when server is start with ENV access and secret keys.
<!--- Not obligatory, but suggest a fix/reason for the bug, -->
<!--- or ideas how to implement the addition or change -->

## Steps to Reproduce (for bugs)
Current minio master and mc master.

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Also changes the behavior of `secretKeyHash` which is
not necessary to be sent over the network, each node
has its own secretKeyHash to validate.

## Motivation and Context
Fixes #3696
Partial(fix) #3700 (More changes needed with some code cleanup)

## How Has This Been Tested?
Manually using docker container.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/config-v13.go [hunk 1]
```diff
@@ -45,13 +45,20 @@ type serverConfigV13 struct {
...
+	}
 	if !isConfigFileExists() {
 		// Initialize server config.
 		srvCfg := &serverConfigV13{}
 		srvCfg.Version = globalMinioConfigVersion
 		srvCfg.Region = globalMinioDefaultRegion
-		srvCfg.Credential = newCredential()
-
+		if globalIsEnvCreds {
+			srvCfg.Credential = creds
```
Comment (krisis) [inline cmd/config-v13.go]: Is it our intention to save the credentials supplied via environment variables in our configuration file?

File: cmd/main.go [hunk 2]
```diff
@@ -174,8 +174,10 @@ func minioInit(ctx *cli.Context) {
 	// Is TLS configured?.
 	globalIsSSL = isSSL()
 
-	// Migrate any old version of config / state files to newer format.
-	migrate()
+	if isConfigFileExists() {
```
Comment (krisis) [inline cmd/main.go]: Can we move the config file existence check into migrate()?

File: cmd/server-main.go [hunk 3]
```diff
@@ -372,8 +367,8 @@ func serverMain(c *cli.Context) {
 		cli.ShowCommandHelpAndExit(c, "server", 1)
 	}
 
-	// Initialization routine, such as config loading, enable logging, ..
-	minioInit(c)
+	// Initialize server.
```
Comment (krisis) [inline cmd/server-main.go]: The comment doesn't say anything about what initialization of server includes.

**Review by krisis — CHANGES_REQUESTED** [review]

File: cmd/config-v13.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/config-v13.go]: Yes when the config doesn't exist.

File: cmd/main.go (hunk 2 above)
Comment (harshavardhana) [reply cmd/main.go]: Can be done.. but i think the flow needs to be better still.

File: cmd/server-main.go (hunk 3 above)
Comment (harshavardhana) [reply cmd/server-main.go]: Done.

File: cmd/main.go (hunk 2 above)
Comment (harshavardhana) [reply cmd/main.go]: Done

**Review by krisis — APPROVED** [review]

**Review by vadmeste — APPROVED** [review]

**Pull request merged**